### PR TITLE
gpuarray tests should pass when floatX=float32 and add this to the daily buildbot

### PR DESCRIPTION
**Pull request by Faruk-Ahmed**
fix gh-4555

### CODE REVIEW

**Comment by nouiz** [comment]
There is this error in travis:

/home/travis/build/Theano/Theano/theano/gpuarray/tests/test_basic_ops.py:92:26: F821 undefined name 'xrange'

File: theano/gpuarray/tests/test_basic_ops.py [hunk 1]
```diff
@@ -89,6 +89,9 @@ def test_all(self):
                 raise SkipTest(skip)
 
             for testname, inputs in iteritems(cases):
+                for _ in xrange(len(inputs)):
```
Comment (nouiz) [inline theano/gpuarray/tests/test_basic_ops.py]: rename xrange to range for python 3 compat.

**Comment by mgermain** [comment]
@Faruk-Ahmed, can you confirm that all the tests under sandbox/gpuarray are all passing  with float32 and float64?
If not can you please list the one that fails.

Thanks

**Comment by nouiz** [comment]
As this fix current problems, I'm merging. But you still need to run again the test to make sure they all pass in float32. Do this with the current master of Theano. thank.

**Comment by Faruk-Ahmed** [comment]
With float32, we have the following error:
======================================================================
FAIL: test_infer_shape (theano.gpuarray.tests.test_extra_ops.TestGpuCumsum)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/u/ahmedfar/Projects/Theano/theano/tensor/tests/test_extra_ops.py", line 157, in test_infer_shape
    self.op_class)
  File "/u/ahmedfar/Projects/Theano/theano/tests/unittest_tools.py", line 249, in _compile_and_check
    assert any(isinstance(t.op, cls) for t in topo_out)
AssertionError

----------------------------------------------------------------------
Some more information:
-> assert any(isinstance(t.op, cls) for t in topo_out)
(Pdb) topo_out
[GpuFromHost<None>(x), GpuReshape{1}(GpuFromHost<None>.0, TensorConstant{(1,) of -1}), GpuCumsum{axis=0}(GpuReshape{1}.0), HostFromGpu(gpuarray)(GpuCumsum{axis=0}.0)]
(Pdb) cls
<class 'theano.tensor.extra_ops.CumsumOp'>


(With float64, all tests pass.)

**Pull request merged**